components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/signup/UserSelfRegistrationManager.java

```diff
@@ -747,4 +749,76 @@ private boolean isFormatCorrect(String regularExpression, String attribute) {
         return m2.matches();
     }
 
+    /**
+     * Pre validate password against the policies defined in the Identity Server during password recovery instance.
+     *
+     * @param confirmationKey Confirmation code for password recovery.
+     * @param password Password to be pre-validated against the policies defined in IS.
+     * @throws IdentityEventException
+     * @throws IdentityRecoveryException
+     */
+    public void preValidatePasswordWithConfirmationKey(String confirmationKey, String password) throws
+            IdentityEventException, IdentityRecoveryException {
+
+        UserRecoveryDataStore userRecoveryDataStore = JDBCRecoveryDataStore.getInstance();
+        UserRecoveryData recoveryData = userRecoveryDataStore.load(confirmationKey);
+        User user = recoveryData.getUser();
+        String userStoreDomain = user.getUserStoreDomain();
+        String username = user.getUserName();
+
+        preValidatePassword(username, password, userStoreDomain);
+    }
+
+    /**
+     * Pre validate the password against the policies defined in the Identity Server.
+     *
+     * @param username Username
+     * @param password Password to be pre-validated against the policies defined in IS.
```

Comment:
Complete the method comment.
If you are adding `.` at the end of param descriptions, add it for all the descriptions

Reply:
fixed in https://github.com/wso2-extensions/identity-governance/pull/355